core/tauri/src/window.rs

```diff
@@ -1395,39 +1395,28 @@ impl<R: Runtime> Window<R> {
     let config_url = manager.get_url();
     let is_local = config_url.make_relative(&current_url).is_some();
 
-    let default_scope = ExternalCommandAccessScope {
-      url: Default::default(),
-      windows: Vec::with_capacity(0),
-      plugins: Vec::with_capacity(0),
-      enable_tauri_api: true,
-    };
-
     let mut scope_not_found_error_message = format!(
       "Scope not defined for window `{}` and URL `{current_url}`",
       self.window.label
     );
     let scope = if is_local {
-      Some(&default_scope)
+      None
     } else {
-      let mut scope = None;
-      let mut found_scope_for_window = false;
-      let mut found_scope_for_url = false;
-      for s in &manager.inner.external_command_access {
-        let matches_window = s.windows.contains(&self.window.label);
-        let matches_url = s.url.matches(current_url.as_str());
-        found_scope_for_window = found_scope_for_window || matches_window;
-        found_scope_for_url = found_scope_for_url || matches_url;
-        if matches_window && matches_url && scope.is_none() {
-          scope.replace(s);
+      match self
+        .ipc_scope()
+        .remote_access_for(&self.window.label, &current_url)
+      {
+        Ok(scope) => Some(scope),
+        Err(e) => {
+          if e.matches_window {
+            scope_not_found_error_message = format!("Scope not defined for URL `{current_url}`");
```

Comment:
@lucasfernog can we point users to the documentation for this new feature, so people know what to search for?

> Scope not defined for URL `{current_url}`. If the domain should be able to access the IPC Layer please refer to  *dangerousRemoteUrlIpcAccess* in the *security* component of the Tauri configuration.